From a8ba9578e361547bc6fdafb2eb9df303a89a5988 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 11 Jan 2017 00:27:59 +0100
Subject: [PATCH 0753/1345] ARM: dts: axp: Add telephony support to DB board

commit  b81e2e342e1e85a6bf2f77b013233eae0b437080 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This commit adds also dual-slic definition under spi@10600 node.

Because telephony interface MPP's overlap ones from SDIO, by default
tdm@b0000 remains disabled. In order to use TDM, enable it and
disable mvsdio@d4000. SLIC modules are enabled in DT, however in case
not used, they will simply be not detected.

Change-Id: Ic1ee847ac44303f1b6cc41e58144bf6774e93c1b
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35874
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-xp-db.dts |   39 ++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm/boot/dts/armada-xp-db.dts b/arch/arm/boot/dts/armada-xp-db.dts
index a339742..b4a856d 100644
--- a/arch/arm/boot/dts/armada-xp-db.dts
+++ b/arch/arm/boot/dts/armada-xp-db.dts
@@ -156,6 +156,19 @@
 				bm,pool-long = <3>;
 			};
 
+			tdm@b0000 {
+				pinctrl-0 = <&tdm_pins>;
+				pinctrl-names = "default";
+				pclk-freq-mhz = <8>;
+				/*
+				 * Node remains disabled, because of
+				 * MPP's overlapping SDIO interface.
+				 * In order to use TDM, disable
+				 * mvsdio@d4000 node, and enable
+				 * tdm@b0000.
+				 */
+			};
+
 			bm@c0000 {
 				status = "okay";
 			};
@@ -267,6 +280,8 @@
 };
 
 &spi0 {
+	pinctrl-0 = <&spi0_pins>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	spi-flash@0 {
@@ -276,4 +291,28 @@
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <20000000>;
 	};
+
+	slic@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "mv_slic";
+		reg = <1>; /* Chip select 1 */
+		spi-max-frequency = <3000000>;
+		spi-1byte-cs;
+		spi-cpol;
+		spi-cpha;
+		slic-id = <0>;
+	};
+
+	slic@2 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "mv_slic";
+		reg = <2>; /* Chip select 2 */
+		spi-max-frequency = <3000000>;
+		spi-1byte-cs;
+		spi-cpol;
+		spi-cpha;
+		slic-id = <1>;
+	};
 };
-- 
1.7.9.5

